tests/TestCase/Database/ConnectionTest.php

```diff
@@ -488,6 +488,25 @@ public function testSimpleTransactions()
     }
 
     /**
+     * Tests that the destructor of Connection does not crash
+     * when transaction is not closed
+     *
+     * @return void
+     */
+    public function testCloseConnectionWithUncommittedTransaction()
+    {
+        $driver = $this->getMockFormDriver();
+        $connection = $this->getMockBuilder('\Cake\Database\Connection')
+            ->setMethods(['connect'])
+            ->setConstructorArgs([['driver' => $driver]])
+            ->getMock();
+        $connection->begin();
+        $this->assertTrue($connection->inTransaction());
+
+        // a warning log will be generated by __destruct of Connection.
```

Comment:
Thanks, I believed that testing the log messages is impossible, but it was wrong.